terraso_backend/apps/graphql/schema/schema.graphql

```diff
@@ -92,6 +94,180 @@ type Query {
   sharedResource(shareUuid: String!): SharedResourceNode
 }
 
+"""Soil ID algorithm queries."""
+type SoilId {
+  locationBasedSoilMatches(latitude: Float!, longitude: Float!): LocationBasedSoilMatches!
+  dataBasedSoilMatches(latitude: Float!, longitude: Float!, data: SoilIdInputData!): DataBasedSoilMatches!
+}
+
+"""A ranked group of soil matches based solely on a coordinate pair."""
+type LocationBasedSoilMatches {
+  matches: [LocationBasedSoilMatch!]!
+}
+
+"""A soil match based solely on a coordinate pair."""
+type LocationBasedSoilMatch {
+  dataSource: String!
+  inMapUnit: Boolean!
```

Comment:
Question (non-blocking for this PR though!): I wasn't entirely clear if this was a boolean or an enum when Courtney mentioned it. Is it just "in soil map unit" or "near soil map unit", and if your location is deemed too far away then it would be one of the "no map data" cases (at least post-Capri)? 

(Please validate this stream of consciousness lol: I was at first like "what should we show in Capri here if it's a 'no map data' case" -- but actually that's not applicable because you can only to the screen that displays that info if you have a non-empty set of soil matches, which requires there being map data ... I think?)

Reply:
i don't fully understand this either, it's on my todo list to look into this a bit more carefully. @CourtneyLee333 could you expand on the intention/source of the requirement for the "In map unit" text in the figma?

Reply:
I think this is the last issue I see with this API, so once this gets squared away I think we're good to merge.